### PR TITLE
Register opset-11 CPU kernel for 'If' op

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: From opset-11, the output shapes of the "then" and "else" branches MAY differ (relaxed from the previous opsets). This change registers a new kernel for opset-11 'If' and validates that the class can handle the new relaxation by adding a test case for the same.

**Motivation and Context**
Make ORT opset-11 compliant

### CODE REVIEW

File: onnxruntime/core/providers/cpu/controlflow/if.cc [hunk 1]
```diff
@@ -46,12 +46,19 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+                                     .TypeConstraint("V", DataTypeImpl::AllTensorTypes()),
+                                   If);
+
 ONNX_CPU_OPERATOR_KERNEL(If,
-                         1,
-                         KernelDefBuilder()
-                             .TypeConstraint("B", DataTypeImpl::GetTensorType<bool>())
-                             .TypeConstraint("V", DataTypeImpl::AllTensorTypes()),
-                         If);
+                        11,
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/controlflow/if.cc]: Comment saying output shape rules were relaxed?

File: onnxruntime/test/providers/cpu/controlflow/if_test.cc [hunk 2]
```diff
@@ -73,7 +74,18 @@ class IfOpTester : public OpTester {
       inputs = {split_input};
       outputs = {&split_out_0, &split_out_1};
 
-      graph.AddNode("split", "Split", "Split into 2", inputs, outputs);
+      auto& split_node = graph.AddNode("split", "Split", "Split into 2", inputs, outputs);
+      if (opset_version_ == 11) {
```
Comment (skottmckay) [inline onnxruntime/test/providers/cpu/controlflow/if_test.cc]: >opset_version_ == 11 [](start = 10, length = 20)

> 10?

**Comment by skottmckay** [comment]
> */

Would be good to update this to mention the second value if opset > 10

---
Refers to: onnxruntime/test/providers/cpu/controlflow/if_test.cc:136 in c29e291. [](commit_id = c29e2919da2073f121390c5460fe22313e6b11ac, deletion_comment = False)

File: onnxruntime/test/providers/cpu/controlflow/if_test.cc (hunk 2 above)
Comment (hariharans29) [reply onnxruntime/test/providers/cpu/controlflow/if_test.cc]: we don't need it for 10 right ? In opset 10, we need to preserve the "even" split, only opset 11 and above supports unequal shapes in both the branches...

File: onnxruntime/core/providers/cpu/controlflow/if.cc (hunk 1 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cpu/controlflow/if.cc]: sure.

File: onnxruntime/test/providers/cpu/controlflow/if_test.cc (hunk 2 above)
Comment (hariharans29) [reply onnxruntime/test/providers/cpu/controlflow/if_test.cc]: Oh I see what you mean now. You wanted `== 11` changed to `> 10`. Will do.

Comment (hariharans29) [reply onnxruntime/test/providers/cpu/controlflow/if_test.cc]: Fixed.

**Comment by hariharans29** [comment]
> > */
> 
> Would be good to update the ascii art for the output from the 'else' subgraph to mention the second value if opset > 10
> 
> Refers to: onnxruntime/test/providers/cpu/controlflow/if_test.cc:136 in [c29e291](https://github.com/microsoft/onnxruntime/commit/c29e2919da2073f121390c5460fe22313e6b11ac). [](commit_id = [c29e291](https://github.com/microsoft/onnxruntime/commit/c29e2919da2073f121390c5460fe22313e6b11ac), deletion_comment = False)

Updated the ascii art. Please check.

File: onnxruntime/core/providers/cpu/controlflow/if.cc (hunk 1 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cpu/controlflow/if.cc]: Added.

**Pull request merged**